### PR TITLE
Feature junit testing

### PR DESCRIPTION
**Pull request by JoelIvanStella**
## Summary
- JUnit testing of controllers, repositories and models
## Trello Card
https://trello.com/c/mo61gIbb

### CODE REVIEW

File: src/test/java/com/wolox/training/controllers/BookControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+    @Test
+    public void whenCreateBook_thenReturnCreatedBook() throws Exception {
+        Mockito.when(mockBookService.create(any(BookVO.class))).thenReturn(oneTestBook);
+        mvc.perform(MockMvcRequestBuilders.post(URL)
+                .content("{}")
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.status().isCreated())
+                .andExpect(MockMvcResultMatchers.content().json("{\n"
+                        + "    \"id\": 0,\n"
```
Comment (torsello) [inline src/test/java/com/wolox/training/controllers/BookControllerTest.java]: Hi Joel, we used to put this JSONs in a file, so you can re-use it. But for me, now you can just replace it with a Constant.

File: src/test/java/com/wolox/training/controllers/UserControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+    @Test
+    public void whenCreateUser_thenReturnCreatedUser() throws Exception {
+        Mockito.when(mockUserService.create(any(UsersVO.class))).thenReturn(oneTestUser);
+        mvc.perform(MockMvcRequestBuilders.post(URL)
+                .content("{}")
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.status().isCreated())
+                .andExpect(MockMvcResultMatchers.content().json("{\n"
+                        + "    \"id\": 0,\n"
```
Comment (torsello) [inline src/test/java/com/wolox/training/controllers/UserControllerTest.java]: In this case, the same. Use constants

**Review by torsello — CHANGES_REQUESTED** [review]
Good job, I leave you some comments

**Review by torsello — APPROVED** [review]
LGTM!

**Review by jdhiguita — APPROVED** [review]
Great job!

**Pull request merged**